### DISCUSSION TITLE
Delays management

### DISCUSSION THREAD

**Discussion by sdelandtsheer** [metrumresearchgroup/mrgsolve#1028] [Q&A]
Hello, I am trying to learn to use the toolbox to illustrate PK/PD simulations to bioinformatics students (has to be in R, hence the switch from our Matlab-based tools). I have a problem when implementing delays, could someone please point me to the respective documentation or example? Thanks a lot in advance

**Comment by kylebaron** [comment]
Hi @sdelandtsheer  - can you give us more information on what sort of delay you are looking to model?

**Reply by sdelandtsheer** [reply]
I am trying to implement the system in this paper: https://pubmed.ncbi.nlm.nih.gov/15771230/
In equation 5 you can see there is the term (t - tau) defining a time-lag.
I have found the "ALAG_CMT" option but I am not sure how to use it.

**Reply by mhismail** [reply]
@sdelandtsheer You can add the lag in $MAIN as ALAG_DEPOT = TAU (you can replace DEPOT with whatever you call the dosing compartment). TAU is the lag time, which you could define under $PARAM.

Note, for this model the authors in the paper write:

> Although typically requiring the use of delay differential equations, such a time-lag can be readily achieved
by delaying the timing of the dosing event in ADAPT 

so, adding the lag time onto PK/dosing input works fine to predict the PD response at time t, but there is no actual lag time in PK. So, if you wanted the PK profile too, you might need to run two sets of simulations (one setting the delay parameter to 0 to simulate PK), or two sets of differential equations, (one set with the delay, and one without).

If you post code, we might be able to help some more.

-Mohamed

**Reply by sdelandtsheer** [reply]
Yes, indeed ALAG seems to implement only a lag on the doses, not the compartments themselves.
I really would like to implement delayed equations, isn't there a workaround?

**Reply by Mutaz94** [reply]
Have you tried adding additional dummy compartments? e.g. `G -> .... -> C` or by adding zero rate input (with controlled duration that can represent the delay).
